Magic should be long.

When neo-express generate a magic bigger than int, neow3j will throw an Exception:

```
Exception in thread "main" com.fasterxml.jackson.databind.JsonMappingException: Numeric value (4232068425) out of range of int
 at [Source: (okio.RealBufferedSource$1); line: 1, column: 125] (through reference chain: io.neow3j.protocol.core.response.NeoGetVersion["result"]->io.neow3j.protocol.core.response.NeoGetVersion$Result["network"])
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394)
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1711)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:290)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
        at io.neow3j.protocol.Neow3j.getNetworkMagicNumber(Neow3j.java:63)
        at io.neow3j.transaction.Transaction.getHashData(Transaction.java:315)
        at io.neow3j.transaction.TransactionBuilder.sign(TransactionBuilder.java:470)
        at info.skyblond.nekohit.example.ContractExample.createWCA(ContractExample.java:38)
        at info.skyblond.nekohit.example.ContractExample.main(ContractExample.java:17)
Caused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (4232068425) out of range of int
 at [Source: (okio.RealBufferedSource$1); line: 1, column: 135]
```

And the value `4232068425` is exactly the private net magic in neo-express config file:
```
{
  "magic": 4232068425,
  "address-version": 53,
  "consensus-nodes": [
    {
      "tcp-port": 50013,
      "ws-port": 50014,
      "rpc-port": 50012,
      "wallet": {
        "name": "node1",
 // ......
```

Looks like neow3j trying to parse magic number into int value, which should be long value.

--- a/core/src/main/java/io/neow3j/protocol/Neow3j.java
+++ b/core/src/main/java/io/neow3j/protocol/Neow3j.java
@@ -62,8 +62,10 @@ public abstract class Neow3j implements Neo, Neow3jRx {
         if (config.getNetworkMagic() == null) {
             config.setNetworkMagic(getVersion().send().getVersion().getNetwork());
         }
+        // transform from long to unsigned int:
+        int networkMagicAsInt = (int) (config.getNetworkMagic() & 0xFFFFFFFFL);
         byte[] array = new byte[4];
-        ByteBuffer.wrap(array).order(ByteOrder.LITTLE_ENDIAN).putInt(config.getNetworkMagic());
+        ByteBuffer.wrap(array).order(ByteOrder.LITTLE_ENDIAN).putInt(networkMagicAsInt);
         return array;
     }
 

--- a/core/src/main/java/io/neow3j/protocol/Neow3jConfig.java
+++ b/core/src/main/java/io/neow3j/protocol/Neow3jConfig.java
@@ -15,7 +15,7 @@ public class Neow3jConfig {
     public static final int MAX_VALID_UNTIL_BLOCK_INCREMENT_BASE = 86400000;
 
     private static byte addressVersion = DEFAULT_ADDRESS_VERSION;
-    private Integer networkMagic = null;
+    private Long networkMagic = null;
     private int blockInterval = DEFAULT_BLOCK_TIME;
     private long maxValidUntilBlockIncrement = MAX_VALID_UNTIL_BLOCK_INCREMENT_BASE / blockInterval;
     private int pollingInterval = DEFAULT_BLOCK_TIME;
@@ -27,7 +27,7 @@ public class Neow3jConfig {
     public Neow3jConfig() {
     }
 
-    public Neow3jConfig(int networkMagic, int blockInterval, int pollingInterval,
+    public Neow3jConfig(long networkMagic, int blockInterval, int pollingInterval,
             long maxValidUntilBlockIncrement, ScheduledExecutorService scheduledExecutorService) {
 
         this.networkMagic = networkMagic;
@@ -120,7 +120,7 @@ public class Neow3jConfig {
      * @return The network's magic number.
      * @see Neow3j#getNetworkMagicNumber()
      */
-    public Integer getNetworkMagic() {
+    public Long getNetworkMagic() {
         return networkMagic;
     }
 
@@ -133,7 +133,7 @@ public class Neow3jConfig {
      * @param magic The network magic number.
      * @return this.
      */
-    public Neow3jConfig setNetworkMagic(int magic) {
+    public Neow3jConfig setNetworkMagic(long magic) {
         networkMagic = magic;
         return this;
     }

--- a/core/src/main/java/io/neow3j/protocol/Neow3jConfig.java
+++ b/core/src/main/java/io/neow3j/protocol/Neow3jConfig.java
@@ -116,6 +116,9 @@ public class Neow3jConfig {
      * The default value is null. Only once {@link Neow3j#getNetworkMagicNumber()} is called for
      * the first time the value is set. This is because the magic number is fetched directly from
      * the neo-node.
+     * <p>
+     * The magic number is represented as an unsigned 32-bit integer on the neo-node. Thus, it's
+     * maximum possible value is 0xffffffff or 2<sup>32</sup>-1.
      *
      * @return The network's magic number.
      * @see Neow3j#getNetworkMagicNumber()

--- a/core/src/main/java/io/neow3j/protocol/Neow3jConfig.java
+++ b/core/src/main/java/io/neow3j/protocol/Neow3jConfig.java
@@ -137,6 +137,10 @@ public class Neow3jConfig {
      * @return this.
      */
     public Neow3jConfig setNetworkMagic(long magic) {
+        if (magic > 0xFFFFFFFFL || magic < 0L)  {
+            throw new IllegalArgumentException("The network magic number must fit into a 32-bit " +
+                    "unsigned integer, i.e., it must be positive and not greater than 0xFFFFFFFF.");
+        }
         networkMagic = magic;
         return this;
     }

--- a/core/src/main/java/io/neow3j/protocol/core/response/NeoGetVersion.java
+++ b/core/src/main/java/io/neow3j/protocol/core/response/NeoGetVersion.java
@@ -28,7 +28,7 @@ public class NeoGetVersion extends Response<NeoGetVersion.Result> {
         private String userAgent;
 
         @JsonProperty("network")
-        private int network;
+        private long network;
 
         public Result() {
         }
@@ -65,7 +65,7 @@ public class NeoGetVersion extends Response<NeoGetVersion.Result> {
             this.userAgent = userAgent;
         }
 
-        public int getNetwork() {
+        public long getNetwork() {
             return network;
         }
 

--- a/core/src/test-integration/java/io/neow3j/protocol/Neow3jReadOnlyIntegrationTest.java
+++ b/core/src/test-integration/java/io/neow3j/protocol/Neow3jReadOnlyIntegrationTest.java
@@ -665,7 +665,7 @@ public class Neow3jReadOnlyIntegrationTest {
         assertThat(versionResult.getNonce(), is(greaterThanOrEqualTo(0L)));
         assertThat(versionResult.getTCPPort(), is(greaterThanOrEqualTo(0)));
         assertThat(versionResult.getWSPort(), is(greaterThanOrEqualTo(0)));
-        assertThat(versionResult.getNetwork(), is(greaterThanOrEqualTo(0)));
+        assertThat(versionResult.getNetwork(), is(greaterThanOrEqualTo(0L)));
     }
 
     // SmartContract Methods

--- a/core/src/test/java/io/neow3j/protocol/core/ResponseTest.java
+++ b/core/src/test/java/io/neow3j/protocol/core/ResponseTest.java
@@ -1413,7 +1413,32 @@ public class ResponseTest extends ResponseTester {
         assertThat(getVersion.getVersion().getWSPort(), is(40334));
         assertThat(getVersion.getVersion().getNonce(), is(224036820L));
         assertThat(getVersion.getVersion().getUserAgent(), is("/Neo:3.0.0/"));
-        assertThat(getVersion.getVersion().getNetwork(), is(769));
+        assertThat(getVersion.getVersion().getNetwork(), is(769L));
+    }
+
+    @Test
+    public void testGetVersion_Network_Long() {
+        buildResponse(
+                "{\n" +
+                        "    \"jsonrpc\": \"2.0\",\n" +
+                        "    \"id\": 1,\n" +
+                        "    \"result\": {\n" +
+                        "        \"tcpport\": 40333,\n" +
+                        "        \"wsport\": 40334,\n" +
+                        "        \"nonce\": 224036820,\n" +
+                        "        \"useragent\": \"/Neo:3.0.0/\",\n" +
+                        "        \"network\": 4232068425\n" +
+                        "    }\n" +
+                        "}"
+        );
+
+        NeoGetVersion getVersion = deserialiseResponse(NeoGetVersion.class);
+        assertThat(getVersion.getVersion(), is(notNullValue()));
+        assertThat(getVersion.getVersion().getTCPPort(), is(40333));
+        assertThat(getVersion.getVersion().getWSPort(), is(40334));
+        assertThat(getVersion.getVersion().getNonce(), is(224036820L));
+        assertThat(getVersion.getVersion().getUserAgent(), is("/Neo:3.0.0/"));
+        assertThat(getVersion.getVersion().getNetwork(), is(4232068425L));
     }
 
     @Test

